### PR TITLE
[posix] add a method to get the infra interface name

### PR DESCRIPTION
**Pull request by zhanglongxia**
When setting the IP forwarding rule on Android platform, it needs to
specify the two interfaces between which IP forwarding is enabled.

This PR adds a method to get the infra interface name for setting IP
forwarding rule.

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

**Review by Irving-cl — APPROVED** [review]
LGTM

File: src/posix/platform/infra_if.hpp [hunk 1]
```diff
@@ -128,6 +128,14 @@ class InfraNetif : public Mainloop::Source, private NonCopyable
                         const otIp6Address &aDestAddress,
                         const uint8_t *     aBuffer,
                         uint16_t            aBufferLength);
+
+    /**
+     * This method gets the infrastructure network interface name.
+     *
+     * @returns The infrastructure network interface name.
+     */
```
Comment (jwhui) [inline src/posix/platform/infra_if.hpp:136]: nit:
```suggestion
     * @returns The infrastructure network interface name.
     *
     */
```

File: src/posix/platform/include/openthread/openthread-system.h [hunk 2]
```diff
@@ -176,6 +176,14 @@ const char *otSysGetThreadNetifName(void);
  */
 unsigned int otSysGetThreadNetifIndex(void);
 
+/**
+ * This method returns the infrastructure network interface name.
+ *
+ * @returns The infrastructure network interface name.
```
Comment (simonlingoogle) [inline src/posix/platform/include/openthread/openthread-system.h:182]:
```suggestion
 * @returns The infrastructure network interface name, or `nullptr` if not specified.
```

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: src/posix/platform/include/openthread/openthread-system.h [hunk 3]
```diff
@@ -176,6 +176,14 @@ const char *otSysGetThreadNetifName(void);
...
+ *
+ */
+const char *otSysGetInfraNetifName(void);
```
Comment (wgtdkp) [inline src/posix/platform/include/openthread/openthread-system.h:185]: The `unsigned int otSysGetThreadNetifIndex(void);` API returns the index of the Thread TUN interface. To make the APIs symmetric, should we unify both return values?

Comment (zhanglongxia) [reply src/posix/platform/include/openthread/openthread-system.h:185]: I think it depends on the requirement. I only need to get the name of the infra network interface now. If we need to get the index of the infra network interface in the future, we can add it later. Thoughts?

**Review by wgtdkp — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**